### ISSUE TITLE
How to get same result on multiple runs ?

### ISSUE THREAD

**Issue by Dongwoo-Im** [XPixelGroup/HAT#21]
Hi, thanks for your sharing and contribution!

I tried to reproduce same result about training loss on my custom dataset, but it didn't work.

So, `I wonder if HAT can return the exactly same result about training loss.`

Any help would be much appreciated, thanks.

My environments
- windows 10
- python : 3.7.13
- pytorch : 1.12.1+cu113
- torchvision : 0.13.1+cu113
- cuda : 11.3
- cudnn : 8.4.1
- basicsr : both 1.3.4.9 and 1.4.2 (latest version)

Methods I tried
- use_hflip = False
- use_rot = False
- use_shuffle = False
- num_worker_per_gpu = 0

**Comment by Dongwoo-Im** [comment]
When I set print_freq to 1, we can check that there's some negligible difference in loss

![image](https://user-images.githubusercontent.com/81875412/192153910-49d12f90-cbf9-4b27-bccb-ed140bf6173b.png)

**Comment by chxy95** [comment]
I do not understand your question. But you can refer to the original training log of HAT for SRx4 on DF2K.
[train_416_train_Final_Model_SRx4_scratch_DF2K_CR3W16P64_500k_B4G8_20220301_002049.log](https://github.com/XPixelGroup/HAT/files/9599677/train_416_train_Final_Model_SRx4_scratch_DF2K_CR3W16P64_500k_B4G8_20220301_002049.log)

**Comment by Dongwoo-Im** [comment]
@chxy95 Thanks for quick response. When I run same yaml file twice on my custom dataset, loss is not same. Maybe this is caused by some non-deterministic algorhitm in HAT. Anyway I want to reproduce same result, but I do not know how to do it.

**Comment by chxy95** [comment]
@Dongwoo-Im I'm not sure the reason for this either. The ONLY random factor in HAT appears to be `drop_path_rate`. It should have no effect on the performance. You can set it to 0 and observe the training process. If it does not work, there may be some random factors that the current version of BasicSR does not take into account.

**Comment by Dongwoo-Im** [comment]
@chxy95 Thanks your advice. I haven't solved yet, but I'll re-open when I get some insights or find solution.
